.github/workflows/main.yml

```diff
@@ -361,6 +361,56 @@ jobs:
           name: 'obs-studio-windows-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
           path: '${{ env.FILE_NAME }}'
 
+  source_build:
+    name: '02 - Source'
+    runs-on: [ubuntu-22.04]
+    if: always()
+    needs: [clang_check]
+    defaults:
+      run:
+        shell: bash
+        working-directory: 'obs-studio'
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          path: 'obs-studio'
+          fetch-depth: 0
+
+      - name: 'Setup environment'
+        id: setup
+        run: |
+          echo "::set-output name=commitHash::$(git rev-parse --short=9 HEAD)"
+
+      - name: 'Create source artifact'
+        if: ${{ success() && github.event_name != 'pull_request' }}
+        run: |
+          outdir=$(pwd)/build
+          mkdir -p ${outdir}
+          prefix=obs-studio-${{ steps.setup.outputs.commitHash }}
+          outfile=${prefix}.tar
+          rm -f ${outfile}
```

Comment:
Check if the file exists before attempting to remove it, don't rely on `-f` - it's an unnecessary call to an external program.

Given that this script only runs on CI, you could even remove this entirely as you're always working on a fresh checkout.

Reply:
although, i don't full y understand the problem with the "unecessary call" (in case the file does exist).

in any case, i'd go for the entire line...